activerecord/test/cases/adapters/mysql2/nested_deadlock_test.rb

```diff
@@ -44,7 +44,11 @@ class Sample < ActiveRecord::Base
         begin
           thread = Thread.new do
             Sample.transaction(requires_new: false) do
+              # any query to make parent transaction dirty
+              # and next nested transaction is going to be a savepoint transaction
+              Sample.take
```

Comment:
Would it be suitable to factor these out as the following?

```ruby
# This should cause the next nested transaction to be a savepoint transaction.
def make_parent_transaction_dirty
  Sample.take
end
```

Or would that defeat "higher-level approach as a real application would use by making a query"?

Reply:
>Or would that defeat "higher-level approach as a real application would use by making a query"?

Not at all, it's actually a good idea, from my point of view extracting it into a method would keep all the context I wanted to share but by making it more readable. Thanks! I'll make a change